js/sdk/src/actyx.ts

```diff
@@ -123,18 +123,43 @@ export const Actyx = {
     const nodeId = await v2getNodeId(opts)
     log.actyx.debug('NodeId call returned:', nodeId)
 
-    if (!nodeId) {
-      // Try connecting to v1 if we failed to retrieve a v2 node id
-      // (Note that if the port is completely unreachable, v2getNodeId will throw an exception and we don’t get here.)
-      log.actyx.debug('NodeId was null, trying to reach V1 backend...')
-      return createV1(opts)
+    const previousInstanceCount = actyxInstanceRegister.count(manifest, opts)
+    if (previousInstanceCount > 0) {
+      console.warn(
+        new Error(
+          `Multiple creations of actyx instance with these parameters are detected! manifest: ${JSON.stringify(
+            manifest,
+          )} opts: ${JSON.stringify(opts)}`,
+        ),
+      )
     }
 
-    log.actyx.debug(
-      'Detected V2 is running, trying to authorize with manifest',
-      JSON.stringify(manifest),
-    )
-    return createV2(manifest, opts, nodeId)
+    // Create actyx instance
+    const actyx = await (() => {
+      if (!nodeId) {
+        // Try connecting to v1 if we failed to retrieve a v2 node id
+        // (Note that if the port is completely unreachable, v2getNodeId will throw an exception and we don’t get here.)
+        log.actyx.debug('NodeId was null, trying to reach V1 backend...')
+        return createV1(opts)
+      }
+
+      log.actyx.debug(
+        'Detected V2 is running, trying to authorize with manifest',
+        JSON.stringify(manifest),
+      )
+      return createV2(manifest, opts, nodeId)
+    })()
+
+    // Patch dispose
+    const oldDispose = actyx.dispose
+    actyx.dispose = () => {
+      const result = oldDispose()
+      actyxInstanceRegister.subOne(manifest, opts)
```

Comment:
Does `oldDispose()` invoke some user callbacks? If so, such callback could try to create a new SDK instance, which in that case would log the “multiple connections” error. If correctness permits it, I’d do the `subOne` before the `oldDispose`.

Reply:
It only calls `MultiplexedWebsocket.dispose` under the hood.
There should be no difference between both orders since this is a sync function.

Reply:
My worry was that there might be an event emitted to user code from that sync function — and if we add such a thing in the future then this problem may well surface.

Reply:
Ah, that's a valid concern

Reply:
fixed https://github.com/Actyx/Actyx/pull/566/commits/fc39630f998fe973970932f29afb1bf9e89934e4